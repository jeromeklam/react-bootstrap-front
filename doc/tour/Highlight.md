## InputLabel

Objet permettant d'ajouter une étape dans le guide de l'application (Aide). 

### Props

#### Obligatoires

| Nom   | Type   | Commentaire       |
| ----- | ------ | ----------------- |
| theme | Chaine | Thème de l'aide   |
| title | Chaine | Texte d'aide      |

#### Optionnels

| Nom      | Type   | Remarque | Commentaire                   |
| -------- | ------ | -------- | ----------------------------- |
| position | Chaine | bottom   | Position de la fenêtre d'aide |

### Exemple

```
<Highlight title="Bouton de recherche" theme="SEARCH">
  <button type="button">Rechercher</button>
</Highlight>
```

### To Do

- Gérer plus de position
- Gérer les objets hors écran ou cachés
- ScrollTo sur de grande pages
