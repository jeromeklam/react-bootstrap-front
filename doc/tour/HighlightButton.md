## InputLabel

Objet permettant d'afficher un après l'autres les aide définies via les objets Highlight.

### Props

#### Obligatoires

| Nom   | Type   | Commentaire       |
| ----- | ------ | ----------------- |
| theme | Chaine | Thème à parcourir |

#### Optionnels

| Nom      | Type   | Remarque | Commentaire                         |
| -------- | ------ | -------- | ----------------------------------- |
| delay    | Entier |          | Délais pour un parcours automatique |

### Exemple

```
<HighlightButton theme="SEARCH" />
```

### To Do

- Gérer le parcours automatique
