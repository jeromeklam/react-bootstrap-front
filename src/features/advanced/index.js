export * from './DefaultForm.jsx';
export { default as DefaultQuickSearch } from './DefaultQuickSearch.jsx';
export { default as HoverObserver } from './HoverObserver.jsx';
export * from './MobileForm.jsx';
export { default as MobileQuickSearch } from './MobileQuickSearch.jsx';
export { default as Portal } from './Portal.jsx';
export * from './ResponsiveConfirm.jsx';
export * from './ResponsiveForm.jsx';
export * from './ResponsiveModal.jsx';
export { default as ResponsiveModalInner } from './ResponsiveModalInner.jsx';
export * from './ResponsiveQuickSearch.jsx';
export { default as WidthObserver } from './WidthObserver.jsx';
export { default as SvgMask } from './SvgMask';
export { default as Follower } from './Follower';
export { default as SvgTimer } from './SvgTimer';
export { default as SvgPlay } from './SvgPlay';
