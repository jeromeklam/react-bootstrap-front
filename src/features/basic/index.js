export { default as Dropdown } from "./Dropdown";
export * from "./InlineInputCheckbox";
export * from "./InlineInputPicker";
export { default as InlineInputSelect } from "./InlineInputSelect";
export * from "./InlineInputText";
export * from "./InlineLabel";
export * from "./InputCheckbox";
export { default as InputDateWeb } from "./InputDateWeb";
export { default as InputDatetime } from "./InputDatetime";
export * from "./InputEmail";
export * from "./InputHidden";
export { default as InputMask } from "./InputMask";
export { default as InputMonetary } from "./InputMonetary";
export * from "./InputPassword";
export { default as InputPicker } from "./InputPicker";
export { default as ButtonPicker } from "./ButtonPicker";
export * from "./InputRadio";
export * from "./InputRandomText";
export { default as InputSelect } from "./InputSelect";
export { default as InputSpin } from "./InputSpin";
export { default as InputStringarray } from "./InputStringarray";
export * from "./InputText";
export { default as InputTextarea } from "./InputTextarea";
export * from "./ResponsiveButton";
export { default as InputCheckList } from "./InputCheckList";
export { default as CalmButton } from "./CalmButton";
export { default as InputGroup } from "./InputGroup";
export { default as InputGroupAppend } from "./InputGroupAppend";
export { default as InputGroupPrepend } from "./InputGroupPrepend";
export { default as InputGroupText } from "./InputGroupText";
export { default as DropdownMenu } from "./DropdownMenu";
export { default as DropdownMenuHeader } from "./DropdownMenuHeader";
export { default as DropdownMenuOption } from "./DropdownMenuOption";
export { default as DropdownMenuDivider } from "./DropdownMenuDivider";
export { default as InputKeyword } from "./InputKeyword";
export { default as InputKeywords } from "./InputKeywords";
export { default as InputWorkload } from "./InputWorkload";
export { default as InputGpsCoords } from "./InputGpsCoords";
export { default as InputPhone } from "./InputPhone";
export { default as Badge } from "./Badge";
export { default as InputDate } from "./InputDate";
export * from "./InputDateNatif";
export { default as InputPhoto } from "./InputPhoto";
export { default as InputBarCode } from "./InputBarCode";
export { default as Scanner } from "./Scanner";
export { default as InputSignature } from "./InputSignature";
export * from "./InputTextMultiline";
export * from "./InputChildren";
