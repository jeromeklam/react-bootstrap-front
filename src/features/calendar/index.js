import EventWrapper from './EventWrapper'
import BackgroundWrapper from './BackgroundWrapper'

export const components = {
  eventWrapper: EventWrapper,
  timeSlotWrapper: BackgroundWrapper,
  dateCellWrapper: BackgroundWrapper,
}

export { default as Calendar } from './Calendar'
export { DateLocalizer } from './localizer'
export { default as momentLocalizer } from './moment'
export { default as globalizeLocalizer } from './globalize'
export { default as dateFnsLocalizer } from './date-fns'
export { default as move } from './utils/move'
export { views as Views, navigate as Navigate } from './utils/constants'
export { default as Agenda } from './Agenda';
export { default as BackgroundCells } from './BackgroundCells';
export { default as DateContentRow } from './DateContentRow';
export { default as DateHeader } from './DateHeader';
export { default as Day } from './Day';
export { default as DayColumn } from './DayColumn';
export { default as EventCell } from './EventCell';
export { default as EventEndingRow } from './EventEndingRow';
export { default as EventRow } from './EventRow';
export { default as EventRowMixin } from './EventRowMixin';
export { default as EventWrapper } from './EventWrapper';
export { default as Header } from './Header';
export { default as IconWrapper } from './IconWrapper';
export { default as Month } from './Month';
export { default as NoopWrapper } from './NoopWrapper';
export { default as Popup } from './Popup';
export { default as ResourceHeader } from './ResourceHeader';
export { default as Selection } from './Selection';
export { default as TimeGrid } from './TimeGrid';
export { default as TimeGridEvent } from './TimeGridEvent';
export { default as TimeGridHeader } from './TimeGridHeader';
export { default as TimeGutter } from './TimeGutter';
export { default as TimeSlotGroup } from './TimeSlotGroup';
export { default as Toolbar } from './Toolbar';
export { default as Week } from './Week';
export { default as WorkWeek } from './WorkWeek';
