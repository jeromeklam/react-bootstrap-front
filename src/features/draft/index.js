export { default as ColorPicker } from "./ColorPicker";
export { default as FontSize } from "./FontSize";
export { default as Inline } from "./Inline";
export { default as InputHtml } from "./InputHtml";