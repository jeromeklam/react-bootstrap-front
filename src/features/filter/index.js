/* MODES */
export const FILTER_MODE_OR = 'OR';
export const FILTER_MODE_AND = 'AND';
/* OPERATORS */
export const FILTER_OPER_LIKE = 'contains';
export const FILTER_OPER_SOUND_LIKE = 'soundex';
export const FILTER_OPER_NOT_LIKE = 'ncontains';
export const FILTER_OPER_START_WITH = 'containsb';
export const FILTER_OPER_END_WITH = 'containse';
export const FILTER_OPER_EQUAL = 'eq';
export const FILTER_OPER_NOT_EQUAL = 'neq';
export const FILTER_OPER_EMPTY = 'empty';
export const FILTER_OPER_NOT_EMPTY = 'nempty';
export const FILTER_OPER_LOWER = 'ltw';
export const FILTER_OPER_LOWER_OR_EQUAL = 'ltwe';
export const FILTER_OPER_LOWER_OR_EQUAL_OR_NULL = 'ltwen';
export const FILTER_OPER_GREATER = 'gt';
export const FILTER_OPER_GREATER_OR_EQUAL = 'gte';
export const FILTER_OPER_GREATER_OR_EQUAL_OR_NULL = 'gten';
export const FILTER_OPER_IN = 'in';
export const FILTER_OPER_NOT_IN = 'nin';
export const FILTER_OPER_BETWEEN = 'between';
/* TYPES */
export const FILTER_TYPE_GROUP = 'GROUP';
export const FILTER_TYPE_ELEM = 'ELEM';
/* SEARCH MODE */
export const FILTER_SEARCH_NONE = 'NONE';
export const FILTER_SEARCH_QUICK = 'QUICK';
export const FILTER_SEARCH_SIMPLE = 'SIMPLE';

export { default as DefaultPanel } from './DefaultPanel.jsx';
export { default as Filter } from './Filter.js';
export { default as FilterBuilder } from './FilterBuilder.jsx';
export { default as InputAutocomplete } from './InputAutocomplete';
export { default as FilterHeader } from './FilterHeader.jsx';
