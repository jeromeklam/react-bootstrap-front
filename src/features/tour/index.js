export { default as Highlight } from './Highlight.jsx';
export { default as HighlightToggler } from './HighlightToggler.jsx';
