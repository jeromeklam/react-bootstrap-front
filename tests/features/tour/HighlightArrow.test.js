import React from 'react';
import renderer from 'react-test-renderer';
import { HighlightArrow } from '../../../src/features/tour';

it('renders with className', () => {
  // First render
  const component = renderer.create(
    <HighlightArrow className="btn bg-secondary" />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});

it('renders with className and inverted', () => {
  // First render
  const component = renderer.create(
    <HighlightArrow inverted className="btn bg-secondary" />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});