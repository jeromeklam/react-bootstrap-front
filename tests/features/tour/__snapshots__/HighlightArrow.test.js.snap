// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with className 1`] = `
<div
  className="btn bg-secondary"
>
  <svg
    viewBox="0 0 18.4 14.4"
  >
    <path
      d="M1.4 7.2h16M7.6 1L1.4 7.2l6.2 6.2"
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
  </svg>
</div>
`;

exports[`renders with className and inverted 1`] = `
<div
  className="btn bg-secondary"
>
  <svg
    viewBox="0 0 18.4 14.4"
  >
    <path
      d="M17 7.2H1M10.8 1L17 7.2l-6.2 6.2"
      fill="none"
      stroke="currentColor"
      strokeLinecap="round"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
  </svg>
</div>
`;
